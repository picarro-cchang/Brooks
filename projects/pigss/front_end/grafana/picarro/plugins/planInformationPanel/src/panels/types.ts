import { any } from "prop-types";

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}
