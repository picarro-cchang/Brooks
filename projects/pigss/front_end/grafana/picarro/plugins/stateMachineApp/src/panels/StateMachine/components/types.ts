export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
