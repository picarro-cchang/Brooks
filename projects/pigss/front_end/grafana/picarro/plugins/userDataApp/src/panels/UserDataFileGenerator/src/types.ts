import { GrafanaTheme } from "@grafana/ui";
import { TimeRange } from "@grafana/data";

export interface DataGeneratorPanelProps {
  timeRange: TimeRange;
}

export interface DataGeneratorLayoutProps {
  theme: GrafanaTheme;
  options: DataGeneratorPanelProps;
}
