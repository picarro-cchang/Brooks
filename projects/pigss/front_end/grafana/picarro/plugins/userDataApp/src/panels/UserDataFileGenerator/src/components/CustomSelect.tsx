import React, { Component } from 'react';
import Select, { components } from 'react-select';

const customStyles = {
  container: provided => ({
    ...provided,
    minWidth: '10%',
  }),
  menu: provided => ({
    ...provided,
    padding: 5,
    backgroundColor: '#262628',
  }),
  option: provided => ({
    ...provided,
    color: '#d8d9da',
    backgroundColor: '#262628',
  }),
  control: provided => ({
    ...provided,
    color: '#d8d9da',
    backgroundColor: '#09090b',
    border: '1px solid #262628',
  }),
  multiValue: provided => ({
    ...provided,
    color: '#d8d9da',
    fontSize: 14,
    backgroundColor: '#09090b',
  }),
  multiValueLabel: provided => ({
    ...provided,
    color: '#d8d9da',
    fontSize: 14,
    backgroundColor: '#09090b',
  }),
};

const Option = props => {
  return (
    <div style={{ display: 'flex' }}>
      <components.Option {...props}>
        <label>
          <input type="checkbox" checked={props.isSelected} onChange={e => null} />
          <span className="checkbox-label">
            {'   '}
            {props.value}
          </span>
        </label>
      </components.Option>
    </div>
  );
};

const ClearIndicator = props => {
  return null;
};

export interface Props {
  passedOptions: [];
  passedOnChange: (keys: any) => void;
  allOption: {
    label: string;
    value: string;
  };
  value: [];
}

export default class CustomSelect extends Component<Props> {
  constructor(props) {
    super(props);
  }
  render() {
    const { passedOptions, passedOnChange, value } = this.props;
    return (
      <Select
        placeholder="Select..."
        components={{ Option, ClearIndicator }}
        styles={customStyles}
        options={passedOptions}
        onChange={passedOnChange}
        value={value}
        // value={passedOptions.find((option: any) => option.value === 'key')}
        isMulti={true}
        closeMenuOnSelect={false}
        hideSelectedOptions={false}
        backspaceRemovesValue={true}
      />
    );
  }
}
