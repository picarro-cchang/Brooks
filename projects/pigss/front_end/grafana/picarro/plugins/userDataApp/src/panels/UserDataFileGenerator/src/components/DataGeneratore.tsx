// import React, { PureComponent } from 'react';
// import DataGeneratorLayout from '../components/DataGeneratorLayout';
// import { shallow, mount } from 'enzyme';
// import { DataGeneratorLayoutProps } from '../types';
// import { TimeRange, dateTime, dateMath, TimeFragment } from '@grafana/data';
// // import { DataGeneratorService } from '../services/__mocks__/DataGeneratorService'
// import 'jest-styled-components';
// import "jest-fetch-mock"

// // const mockgetFile = jest.fn((fileName) => {return fileName;});
// // const apiLoc = `${window.location.hostname}:8000/controller`;
// // const socketURL = `ws://${apiLoc}/ws`;


// const defaultProps: DataGeneratorLayoutProps = {
//     theme: null,
//     options: {
//         timeRange: {
//             from: dateTime("2020-02-11T14:31:13.973Z"),
//             to: dateTime("2020-02-11T20:31:13.973Z"),
//             raw: { from: 'now-6h' as TimeFragment, to: 'now' as TimeFragment },
//         }
//     }
// };

// describe('<DataGeneratorLayout />', () => {
//   const wrapper = shallow(<DataGeneratorLayout {...defaultProps}/>);
//   // const instance = wrapper.instance() as DataGeneratorLayout;
// //   const server = new WS(socketURL);
// //   const client = new WebSocket(socketURL);

//   it('Renders correctly', () => {
//     expect(wrapper).toMatchSnapshot();
//   });
  
// //   it('Generate button disabled until selections are made', () => {});

//   // it('DataService generateFile functionality', () => {
//       // wrapper.setState({
//       //     ...defaultProps,
//       //     timeRange: {
//       //         from: dateTime().subtract(6, 'h'),
//       //         to: dateTime(),
//       //         raw: { from: 'now-6h' as TimeFragment, to: 'now' as TimeFragment }
//       //     },
//       //     keys: [{value: "WarmBoxTemp", label: "WarmBoxTemp"}],
//       //     analyzers: [{value: "AMADS3001", label: "AMADS3001"}],
//       //     ports: [{value: "All", label: "All"}]
//       // });
//       // const { from, to } = {
//       //   from: dateMath.parse(wrapper.state()["timeRange"].raw.from),
//       //   to: dateMath.parse(wrapper.state()["timeRange"].raw.to),
//       //   raw: wrapper.state()["timeRange"].raw
//       // } as TimeRange;
//       // // @ts-ignore
//       // let fromTime = from._d.getTime();

//       // // @ts-ignore
//       // let toTime = to._d.getTime();

//       // const queryParams = {
//       //   ports: wrapper.state()["ports"],
//       //   from: fromTime * 1000000,
//       //   to: toTime * 1000000,
//       //   keys: wrapper.state()["keys"],
//       //   analyzers: wrapper.state()["analyzers"]
//       // };

//       // // return DataGeneratorService.generateFile(queryParams).then(response => {
//       // //   expect(response).toEqual({"filename": "picarro-02-12-2020_095753-02-12-2020_155753.csv"})
//       // // })

// // });
// });

// //{"filename": "picarro-02-12-2020_095753-02-12-2020_155753.csv"}