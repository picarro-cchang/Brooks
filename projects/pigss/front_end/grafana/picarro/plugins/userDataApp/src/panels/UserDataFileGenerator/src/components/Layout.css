.file-item {
  width: auto;
  padding: 8px;
  margin-bottom: 2px;
}

.file-item:hover {
  cursor: pointer;
  color: #3bd2f8;
  text-decoration: underline;
  background-color: #fffdfd14;
}

/* Fix for grafana panel scrolling issue */
/* Have to figure out what's wrong with grafana or the panel! */
.panel-content {
  overflow: auto;
}

.time-picker-container .navbar-button--zoom {
  display: none;
}
