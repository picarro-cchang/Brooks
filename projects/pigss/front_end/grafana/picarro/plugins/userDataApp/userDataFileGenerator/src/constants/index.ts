import { TimeOption, TimeRange, dateTime, TimeFragment } from '@grafana/data';
import { DataGeneratorPanelProps } from 'types';

export const DEFAULT_TIME_RANGE: TimeRange = {
  from: dateTime().subtract(6, 'h'),
  to: dateTime(),
  raw: { from: 'now-6h' as TimeFragment, to: 'now' as TimeFragment },
};

export const DEFAULT_DATA_GENERATOR_PROPS: DataGeneratorPanelProps = {
  timeRange: DEFAULT_TIME_RANGE,
};

export const DEFAULT_TIME_OPTIONS: TimeOption[] = [
  { from: 'now-5m', to: 'now', display: 'Last 5 minutes', section: 3 },
  { from: 'now-15m', to: 'now', display: 'Last 15 minutes', section: 3 },
  { from: 'now-30m', to: 'now', display: 'Last 30 minutes', section: 3 },
  { from: 'now-1h', to: 'now', display: 'Last 1 hour', section: 3 },
  { from: 'now-3h', to: 'now', display: 'Last 3 hours', section: 3 },
  { from: 'now-6h', to: 'now', display: 'Last 6 hours', section: 3 },
  { from: 'now-12h', to: 'now', display: 'Last 12 hours', section: 3 },
  { from: 'now-24h', to: 'now', display: 'Last 24 hours', section: 3 },
];

export const URL = {
  GET_SAVED_FILES: `http://${window.location.hostname}:8000/grafana_data_generator/api/getsavedfiles`,
  GET_FIELD_KEYS: `http://${window.location.hostname}:8000/grafana_data_generator/api/getkeys`,
  GET_FILE: `http://${window.location.hostname}:8000/grafana_data_generator/api/getfile`,
  GENERATE_FILE: `http://${window.location.hostname}:8000/grafana_data_generator/api/generatefile`,
};
