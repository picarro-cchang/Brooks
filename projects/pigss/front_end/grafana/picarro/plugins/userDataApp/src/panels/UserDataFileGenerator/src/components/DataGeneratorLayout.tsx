import React, { PureComponent, Fragment } from 'react';
import { TimeRange, TIME_FORMAT, dateTime, dateMath } from '@grafana/data';
import { TimePicker, FormLabel, PanelOptionsGroup, Select, Button } from '@grafana/ui';

import { notifyError, notifySuccess } from '../utils/Notifications';
import { DataGeneratorLayoutProps } from '../types';
import { DEFAULT_TIME_OPTIONS } from '../constants';
import { DataGeneratorService } from '../services/DataGeneratorService';

import './Layout.css';

interface Props extends DataGeneratorLayoutProps { }

const labelWidth_6 = 6;
const labelWidth_7 = 7;
const selectWidth = 12;

export default class DataGeneratorLayout extends PureComponent<Props, any> {
  constructor(props: Props) {
    super(props);
    this.state = {
      timeRange: this.props.options.timeRange,
      keys: [],
      keyOptions: [],
      files: [],
    };
  }

  // Code to download the file
  downloadData = (blob: any, fileName: string) => {
    const a = document.createElement('a');

    document.getElementById('file-list').appendChild(a);
    // @ts-ignore
    a.style = 'display: none';
    const url = window.URL.createObjectURL(blob);
    a.href = url;
    a.download = fileName;
    a.click();
    notifySuccess(`File successfully downloaded in Downloads folder.`);
    window.URL.revokeObjectURL(url);
  };

  generateFile = () => {
    const { timeRange, keys } = this.state;
    const {from, to } = {
      from: dateMath.parse(timeRange.raw.from),
      to: dateMath.parse(timeRange.raw.to),
      raw: timeRange.raw
    } as TimeRange;
    
    // @ts-ignore
    let fromTime = from._d.getTime();
    // @ts-ignore
    let toTime = to._d.getTime();

    if (fromTime === toTime || keys.length === 0) {
      notifyError('Invalid Query parameters');
      return;
    }

    const queryParams = {
      from: fromTime * 1000000,
      to: toTime * 1000000,
      keys: keys,
    };

    // Call generate file api
    DataGeneratorService.generateFile(queryParams).then((response: any) => {
      response.json().then((data: any) => {
        // call getFile to download the data
        if (data.filename !== undefined) {
          this.getFileNames();
          this.getFile(data.filename);
        } else {
          if (data !== undefined && data.hasOwnProperty('message')) {
            notifyError(data.message);
          }
        }
      });
    });
  };

  getFile = (fileName: string) => {
    DataGeneratorService.getFile(fileName).then((response: any) => {
      response.blob().then((data: any) => {
        this.downloadData(data, fileName);
      });
    });
  };

  getFileNames = () => {
    // Get file names
    DataGeneratorService.getSavedFiles().then((response: any) => {
      response.json().then((files: any) => {
        this.setState(() => {
          return files;
        });
      });
    });
  };

  onDateChange = (timeRange: TimeRange) => {
    this.setState({ timeRange });
  };

  onKeysChange = (keys: any) => {
    this.setState(() => {
      return { keys };
    });
  };

  componentWillMount() {
    // Get file names
    this.getFileNames();

    // Get Key Options
    DataGeneratorService.getKeys().then((response: any) => {
      response.json().then((data: any) => {
        const keyOptions = data.keys.map((x: string) => {
          return {
            value: x,
            label: x,
          };
        });
        this.setState({ keyOptions: keyOptions });
      });
    });
  }

  render() {
    const { files, timeRange, keyOptions } = this.state;

    const styleObj = {
      overflow: 'scroll !important',
      height: 'inherit',
    };

    if (typeof timeRange.from === 'string') {
      timeRange.from = dateTime(timeRange.from);
    }
    if (typeof timeRange.to === 'string') {
      timeRange.to = dateTime(timeRange.to);
    }

    const fileItems = files.map((fn: string, i: number) => {
      return (
        <li key={fn} className="file-item" value={fn} onClick={() => this.getFile(fn)}>
          {fn}
        </li>
      );
    });

    return (
      <Fragment>
        <h3 className="text-center">User Data File Generator</h3>
        <PanelOptionsGroup title="Generate New File">
          <div className="row">
            <div className="gf-form col-md-3 col-sm-12">
              <FormLabel width={labelWidth_6}>Species</FormLabel>
              <Select
                width={selectWidth}
                options={keyOptions}
                onChange={this.onKeysChange}
                value={keyOptions.find((option: any) => option.value === 'key')}
                isMulti={true}
                backspaceRemovesValue
              />
            </div>
            <div className="gf-form col-md-6 col-sm-12 time-picker-container">
              <FormLabel width={labelWidth_7}>Time Range</FormLabel>
              <TimePicker
                timeZone="browser"
                selectOptions={DEFAULT_TIME_OPTIONS}
                onChange={this.onDateChange}
                value={this.state.timeRange}
                onMoveBackward={() => console.log('Move Backward')}
                onMoveForward={() => console.log('Move forward')}
                onZoom={() => console.log('Zoom')}
              />
            </div>

            <div className="gf-form col-md-3 col-sm-12">
              <Button size="md" variant="primary" value="Generate" onClick={this.generateFile} disabled={!this.state.keys.length}>
                Generate
              </Button>
            </div>
          </div>
        </PanelOptionsGroup>
        <PanelOptionsGroup title="Recently Generated Files">
          <div className="row" style={styleObj}>
            <div className="gf-form col-md-12 col-sm-12">
              <ul id="file-list" style={{ listStyle: 'none' }}>
                {fileItems}
              </ul>
            </div>
          </div>
        </PanelOptionsGroup>
      </Fragment>
    );
  }
}
