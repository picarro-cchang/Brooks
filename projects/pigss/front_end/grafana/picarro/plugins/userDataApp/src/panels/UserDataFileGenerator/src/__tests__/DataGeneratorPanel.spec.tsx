import React from "react";
import { DataGeneratorPanel } from "./../components/DataGeneratorPanel";
import { shallow } from "enzyme";
import { DataGeneratorPanelProps } from "../types";
import { PanelProps, ThemeContext } from "@grafana/ui";
import { GrafanaTheme } from "@grafana/ui";
import { DEFAULT_TIME_RANGE } from "./../constants/index";

jest.mock("./../services/DataGeneratorService.ts");
jest.mock("./../utils/Notifications.ts");

const defaultProps: PanelProps<DataGeneratorPanelProps> = {
  timeRange: DEFAULT_TIME_RANGE,
  id: 2,
  data: null,
  timeZone: "browser",
  options: null,
  onOptionsChange: () => {},
  renderCounter: 0,
  transparent: true,
  width: 1692,
  height: 812,
  replaceVariables: null,
  onChangeTimeRange: () => {}
};
const theme = { theme: "Grafana Dark" };
describe("<DataGeneratorPanel/>", () => {
  const wrapper = shallow(<DataGeneratorPanel {...defaultProps} {...theme} />);

  it("Create Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
