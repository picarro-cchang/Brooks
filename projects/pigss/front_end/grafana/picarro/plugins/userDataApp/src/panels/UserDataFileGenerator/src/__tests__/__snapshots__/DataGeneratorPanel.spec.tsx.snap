// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataGeneratorPanel/> Create Snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
